common/persistence/class.PhpFileDriver.php

```
@@ -304,12 +448,61 @@ protected function sanitizeReadableFileName($key) {
      * 
      * @param string $key
      * @param mixed $value
+     *
      * @return string
+     *
+     * @throws \common_exception_Error
      */
-    protected function getContent($key, $value) {
+    protected function getContent($key, $value)
+    {
         return $this->humanReadable
             ? "<?php return ".common_Utils::toHumanReadablePhpString($value).";".PHP_EOL
             : "<?php return ".common_Utils::toPHPVariableString($value).";";
     }
 
+    /**
+     * Sets the requested value to the cache.
+     *
+     * @param $id
+     * @param $value
+     */
+    protected function setToCache($id, $value)
+    {
+        $this->cache[$id] = $value;
+    }
+
+    /**
+     * Returns the requested element from cache or FALSE if it does not exist.
+     *
+     * @param $id
+     *
+     * @return bool|mixed
+     */
+    protected function getFromCache($id)
```

Please throw an exception if the value does not exist